airflow/kubernetes/worker_configuration.py

```diff
@@ -389,11 +386,14 @@ def make_pod(self, namespace, worker_uuid, pod_id, dag_id, task_id, execution_da
             envs=self._get_environment(),
             node_selectors=self.kube_config.kube_node_selectors,
             service_account_name=self.kube_config.worker_service_account_name,
-        )
+            pod_template_file_or_string=self.kube_config.pod_template_file
+        ).gen_pod()
 
-        pod = pod_generator.gen_pod()
-        pod.spec.containers[0].env_from = pod.spec.containers[0].env_from or []
-        pod.spec.containers[0].env_from.extend(self._get_env_from())
-        pod.spec.security_context = self._get_security_context()
+        if self.kube_config.pod_template_file:
```

Comment:
Is there any reason why we couldn't have both file and config options work? Do you just think it would be to confusing?

If both aren't supported then I would feel happier if we validated the config at `airflow` start time so and raised an error then if conflicting config options are given.

Reply:
Agreed we should validate at startup

Reply:
The logic behind file and config options would be fairly complex. Would config override file or file override config? That would also further the complexity of this PR.
I added some fairly meager validation [here](https://github.com/apache/airflow/blob/06e942e2c0a4009e731c33bf7c4f79ace0175a5a/airflow/executors/kubernetes_executor.py#L230) essentially just allowing the kube_configuration to be empty if `pod_template_file` is passed. The other alternative would be erroring on anything that is non-empty in the kube_config, which seems overly strict.

Reply:
The behavoir I was wanting to guard against is when `pod_template_file` is set in the config, and then someone comes along and sets, say `AIRFLOW__KUBERNETES__WORKER_CONTAINER_TAG` thinking "I know, I just want to test this one tag, that's easier than editing the file".

I'd like that either to work (where pod template file is base, and then any config option set takes precenedece over the file), or for it to error. The thing I don't want to happen is the setting to be ignored silently causing hard-to-debug "why isn't that config option working!" rage.

Reply:
@davlum I'd like to see more validation please -- ideally if _any_ of the other settings are set/have a non-default value and `pod_template_file` is I'd like it to throw an exception.